File: Afrid/front-end/src/HotelService.js
import axios from "axios";

const BASE_URL = "http://localhost:8084/api/v1/hotel";

class HotelService {
  getAllHotels() {
    return axios.get(`${BASE_URL}/getAllHotel`);
  }

  addHotel(hotel) {
    return axios.post(`${BASE_URL}/save`, hotel);
  }

  updateHotel(hotelId, hotel) {
    return axios.put(`${BASE_URL}/update/${hotelId}`, hotel);
  }

  deleteHotel(hotelId) {
    return axios.delete(`${BASE_URL}/deletehotel/${hotelId}`);
  }
}

export default new HotelService();

File: Afrid/back-end/Management/src/main/java/com/Project/Management/Service/HotelService.java
package com.Project.Management.Service;

import com.Project.Management.DTO.HotelDTO;
import com.Project.Management.DTO.HotelSaveDTO;
import com.Project.Management.Entity.Hotel;

import java.util.List;

public interface HotelService {
    Hotel addHotel(HotelSaveDTO hotelSaveDTO);

    List<HotelDTO> getAllHotels();

    Hotel updateHotel(int hotelid, Hotel hotel);

    boolean deleteHotel(int id);

}
